test/object-store/sync/flx_sync.cpp

```diff
@@ -730,6 +730,7 @@ TEST_CASE("flx: client reset", "[sync][flx][client reset][baas]") {
                 REQUIRE(mode == ClientResyncMode::Recover);
                 auto subs = local_realm->get_latest_subscription_set();
                 subs.get_state_change_notification(sync::SubscriptionSet::State::Complete).get();
+                subs.refresh();
```

Comment:
This test was relying on wait_for_upload_completion() waiting for subscription changes to be uploaded (which it would only sometimes do and wasn't actually guaranteed), which happened to result in the subscription state being Complete before the call to get_state_change_notification() so this worked without the refresh.

Reply:
has this been flaky lately or something? i think waiting_for_upload_completion() _used_ to guarantee this, but maybe that's changed from under me.

Reply:
`check_for_upload_completion()` has had specific logic to report completion even if there's unuploaded changesets as long as it had scanned all of the changesets (i.e. as long as all of the remaining ones are empty or from the server) since 2018, and I'm pretty sure there was equivalent behavior achieved differently before that.

The change in functionality that broke this test is that we don't scan the changesets to see if any needed to be uploaded until after the first DOWNLOAD is received, so previously empty changesets made wait_for_uploads() wait for the first DOWNLOAD message and now it doesn't. It's probably possible to preserve that behavior, but it seems really weird and inconsistent (particularly because the presence of empty changesets may not be directly related to anything the developer did).

Either way, the test was incorrect; it should either be waiting on the state change notification and then calling refresh or simply asserting the state without waiting. Waiting then asserting without the refresh in between doesn't really make any sense.